watson/cli.py

```diff
@@ -564,50 +567,66 @@ def log(watson, current, from_, to, projects, tags, year, month, week, day):
                               cur['tags'], id="current")
 
     span = watson.frames.span(from_, to)
-    frames_by_day = sorted_groupby(
-        watson.frames.filter(
-            projects=projects or None, tags=tags or None, span=span
-        ),
-        operator.attrgetter('day'), reverse=True
+    filtered_frames = watson.frames.filter(
+        projects=projects or None, tags=tags or None, span=span
     )
 
-    lines = []
+    if format_json:
+        log = [
+            {
+                'id': frame.id,
+                'start': frame.start.isoformat(),
+                'stop': frame.stop.isoformat(),
+                'project': frame.project,
+                'tags': frame.tags,
+            }
+            for frame in filtered_frames
+        ]
+        click.echo(json.dumps(log, indent=4, sort_keys=True))
+    else:
```

Comment:
I think it would be cleaner to return in the if, so that we can avoid the else and the indentation.